File: src/tmp/Python/tree/search_in_a_binary_search_tree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def searchBST(root: TreeNode, val: int) -> TreeNode:
    while root is not None:
        if root.val == val:
            break
        elif root.val < val:
            root = root.right
        else:
            root = root.left
    
    return root

if __name__ == "__main__":
    print(searchBST())

File: src/tmp/Java/src/com/dbc/algorithm/Tree/SearchInABinarySearchTree.java
package com.dbc.algorithm.Tree;

public class SearchInABinarySearchTree {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode() {
        }

        TreeNode(int val) {
            this.val = val;
        }

        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    public TreeNode searchBST(TreeNode root, int val) {
        while (root != null) {
            if (root.val == val)
                break;
            else if (root.val < val) {
                root = root.right;
            } else {
                root = root.left;
            }
        }
        return root;
    }
}
